common/src/main/java/org/keycloak/common/Profile.java

```diff
@@ -95,6 +119,8 @@ private Profile() {
 
         for (Feature f : Feature.values()) {
             Boolean enabled = config.getConfig(f);
+            Type type = product.equals(ProductValue.RHSSO) ? f.getTypeProduct() : f.getTypeProject();
```

Comment:
Is it possible to remove field "defaultType" and methods "setDefaultType" and "getType" from the Feature enum? AFAIK the mutable enums is the antipattern - see for example https://rules.sonarsource.com/java/RSPEC-3066 . Hopefully all occurences of "f.getType()" in the further code can be replaced simply with the "type", which was retrieved on line 122? For exampe:
```
switch (type) {
```

instead of:
```
switch (f.getType()) {
```

etc

Reply:
@mposolda I see, it's good to know that. I've just wanted to ensure the `getType()` method really works. Thanks!